CHANGELOG.md

```diff
@@ -1,5 +1,126 @@
+    .concurrency(3)
+    .run();
+```
+
+For details, see [Concurrent Test Execution](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/concurrent-test-execution.html).
+
+#### :gear: Further improvements in automatic waiting mechanism ([#1521](https://github.com/DevExpress/testcafe/issues/1521))
+
+We have enhanced the waiting mechanism behavior in certain scenarios where you still used to need `wait` actions.
+Now automatic waiting is much smarter and chances that you need to wait manually are diminished.
+
+#### :gear: Selector's withAttribute method supports search by strict match ([#1548](https://github.com/DevExpress/testcafe/issues/1548]))
+
+When you provide string arguments to the `withAttribute` method, the selector searches
+for an element whose attribute name and value strictly match the given parameters.
+
+```js
+const el = Selector('.container').withAttribute('width', '200');
+```
+
+#### :gear: User roles preserve the local storage ([#1454](https://github.com/DevExpress/testcafe/issues/1454))
+
+TestCafe now saves the local storage state when switching between roles, so that when you switch back you get the same local storage content you left.
+
+This is useful for testing websites that perform authentication via local storage instead of cookies.
+
+#### Miscellaneous
+
+* `StorageSandbox` is shared to the client API with `clear` and `lock` methods added. ([testcafe-hammerhead/#1189](https://github.com/DevExpress/testcafe-hammerhead/issues/1189))
+* The number of failed cookie-sync messages is now limited. ([testcafe-hammerhead/#1193](https://github.com/DevExpress/testcafe-hammerhead/issues/1193))
+* `Hammerhead.backupStorages` and `Hammerhead.restoreStorages` methods are added to the client API([testcafe-hammerhead/#1125](https://github.com/DevExpress/testcafe-hammerhead/issues/1125))
+* The `GetEventListeners` method is implemented in `sandbox.event.listeners`. ([testcafe-hammerhead/#1238](https://github.com/DevExpress/testcafe-hammerhead/issues/1238))
+* JavaScript expressions now supported in raw API selectors. ([#1466](https://github.com/DevExpress/testcafe/issues/1466))
+
+### Bug Fixes
+
+* Authentication with user roles works correctly in SPAs. ([#1502](https://github.com/DevExpress/testcafe/issues/1502))
+* Description for the `path` parameter of the `t.takeScreenshot` action has been corrected ([#1515](https://github.com/DevExpress/testcafe/issues/1515))
+* Local storage is now cleaned appropriately after the test run.([#1546](https://github.com/DevExpress/testcafe/issues/1546))
+* TestCafe now checks element visibility with a timeout when the target element's `style.top` is negative ([#1185](https://github.com/DevExpress/testcafe/issues/1185))
+* Fetching an absolute CORS URL now works correctly. ([#1629](https://github.com/DevExpress/testcafe/issues/1629))
+* Fixed a bug that led to an error in test code. ([#1442](https://github.com/DevExpress/testcafe/issues/1442))
```

Comment:
Let's change to more sense description: `Add partial support for proxing live node collections (GetElementsByTagName method)`